bokehjs/src/lib/models/plots/plot_canvas.ts

```diff
@@ -653,6 +657,7 @@ export class PlotCanvasView extends DOMView {
 
     this.connect(this.model.plot.properties.renderers.change, () => this.build_levels())
     this.connect(this.model.plot.toolbar.properties.tools.change, () => { this.build_levels(); this.build_tools() })
+    this.connect(this.model.plot.toolbar.properties.visible.change, () => { this.build_levels(); this.build_tools() })
```

Comment:
What is the need to rebuild the tools here? Does a request to render not suffice?

Reply:
I was pretty all over the place at the beginning of this PR. This isn't needed.